File: Rithvik/insertInterval.py
class Solution(object):
    def insert(self, intervals, newInterval):
        """
        :type intervals: List[List[int]]
        :type newInterval: List[int]
        :rtype: List[List[int]]
        """
        modIntervals = []
        inserted = False
        for interval in intervals:
            if interval[0] > newInterval[0] and inserted == False:
                modIntervals.append(newInterval)
                inserted = True
            modIntervals.append(interval)
        if not inserted:
            modIntervals.append(newInterval)
        mergedIntervals = [modIntervals[0]]
        if len(modIntervals) == 1:
            return mergedIntervals
        currIndex = 0
        for interval in modIntervals[1:]:
            if interval[0] <= mergedIntervals[currIndex][1]:
                mergedIntervals[currIndex][1] = max(interval[1], mergedIntervals[currIndex][1])
            else:
                mergedIntervals.append(interval)
                currIndex += 1
        return mergedIntervals
        

File: Christine/May/InsertInterval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]> inserted = new ArrayList<>();

        int i = 0;

        while ((i < intervals.length) && (intervals[i][1] < newInterval[0])) {
            inserted.add(intervals[i]);

            i++;
        }

        int first = newInterval[0];
        int second = newInterval[1];

        while ((i < intervals.length) && (intervals[i][0] <= second)) {
            first = (first < intervals[i][0] ? first : intervals[i][0]);
            second = (second > intervals[i][1] ? second : intervals[i][1]);

            i++;
        }

        inserted.add(new int[]{first, second});

        while (i < intervals.length) {
            inserted.add(intervals[i]);

            i++;
        }
        
        return inserted.toArray(new int[inserted.size()][]);
    }
}
